### ISSUE TITLE
Can't install app on tablet running Android 5.0.2.

### ISSUE THREAD

**Issue by eggpi** [jitsi/jitsi-meet#1736]
Hi,

I'm trying to install the Jitsi Meet app on a [Samsung Galaxy Tab 4](https://en.wikipedia.org/wiki/Samsung_Galaxy_Tab_4_10.1) running Android 5.0.2.

Even though the Play Store app tells me the minimum required version is 4.1 (which makes sense I suppose, given the dependence on React Native), I still get a notice that the app is incompatible with my device, so I can't install it. Any ideas why? Are tablets supposed to be supported at all?

Thanks!

**Comment by saghul** [comment]
Sorry, I have no idea :-( Does the error message give any details? An error code perhaps? Tablets should be supported, yes.

**Comment by eggpi** [comment]
Not much I'm afraid :( Looks like the web version of Play store [used to surface more information](https://lifehacker.com/google-play-shows-you-why-an-app-is-incompatible-with-y-496328577) in this case, but that's I don't see it anymore. All I get is "This item is not compatible with your device".

The one thing that perhaps stands out is the fact that, if I attempt to install it on another device attached to my account (a phone), it requests a phone-specific permission (`android.permission.READ_PHONE_STATE` I believe). This tablet has no SIM, so perhaps it can't satisfy that permission and is then considered incompatible?

I don't see that permission in the AndroidManifest.xml in this repository though, so I wonder if the version in the Play Store is outdated or whether this permission comes from some dependency...

**Comment by eggpi** [comment]
Oh, I think I have a better hint now. I finally have the tablet near me so I'm able to do more effective troubleshooting.

So following my suspicion that this is due to some [device feature](https://developer.android.com/guide/practices/compatibility.html) that I might be missing, I ended up building a local apk and looking at the device features it requires:

```
$ aapt dump badging android/app/build/outputs/apk/debug/app-debug.apk
feature-group: label=''
  uses-gl-es: '0x20000'
  uses-feature: name='android.hardware.camera'
  uses-feature: name='android.hardware.camera.autofocus'
  uses-feature: name='android.hardware.bluetooth'
  uses-implied-feature: name='android.hardware.bluetooth' reason='requested android.permission.BLUETOOTH permission, and targetSdkVersion > 4'
  uses-feature: name='android.hardware.microphone'
  uses-implied-feature: name='android.hardware.microphone' reason='requested android.permission.RECORD_AUDIO permission'
  uses-feature: name='android.hardware.touchscreen'
  uses-implied-feature: name='android.hardware.touchscreen' reason='default feature for all apps'
```

Now, looking at the features in the tablet:

```
$ adb shell pm list features
feature:reqGlEsVersion=0x30000
feature:android.hardware.audio.output
feature:android.hardware.bluetooth
feature:android.hardware.bluetooth_le
feature:android.hardware.camera
feature:android.hardware.camera.any
feature:android.hardware.camera.front
feature:android.hardware.faketouch
feature:android.hardware.location
feature:android.hardware.location.gps
feature:android.hardware.location.network
feature:android.hardware.microphone
feature:android.hardware.screen.landscape
feature:android.hardware.screen.portrait
feature:android.hardware.sensor.accelerometer
feature:android.hardware.touchscreen
feature:android.hardware.touchscreen.multitouch
feature:android.hardware.touchscreen.multitouch.distinct
feature:android.hardware.touchscreen.multitouch.jazzhand
feature:android.hardware.usb.accessory
feature:android.hardware.usb.host
feature:android.hardware.wifi
feature:android.hardware.wifi.direct
feature:android.software.app_widgets
feature:android.software.backup
feature:android.software.connectionservice
feature:android.software.device_admin
feature:android.software.home_screen
feature:android.software.input_methods
feature:android.software.live_wallpaper
feature:android.software.print
feature:android.software.sip
feature:android.software.sip.voip
feature:android.software.voice_recognizers
feature:android.software.webview
feature:com.samsung.android.feature.sdl.2101
feature:com.sec.android.mdm
feature:com.sec.android.smartface.smart_stay
feature:com.sec.feature.call_vt_support
feature:com.sec.feature.cover.flip
feature:com.sec.feature.minimode
feature:com.sec.feature.motionrecognition_service
feature:com.sec.feature.multiwindow
feature:com.sec.feature.multiwindow.bezelui
feature:com.sec.feature.multiwindow.commonui
feature:com.sec.feature.multiwindow.freestyle
feature:com.sec.feature.multiwindow.multiwindowlaunch
feature:com.sec.feature.multiwindow.scalewindow
feature:com.sec.feature.wfd_support
```

It looks like the tablet is missing **android.hardware.camera.autofocus**, which, if I'm reading the code right, we [explicitly require](https://github.com/jitsi/jitsi-meet/blob/4efbbe14b100ce17c6ea82824d205576373ef732/android/sdk/src/main/AndroidManifest.xml#L13).

Unfortunately adding or removing that requirement makes no difference on the behavior I get when running `react-native run-android`: the installation always works, but then the app itself raises an exception (`Focus error, retry after 1000`) when trying to join a chat.

I can paste the full exception if you're interested, but I guess the important thing is that I don't know how to reproduce whatever checks the Play Store is doing, so I'm not 100% sure the device feature is the culprit, though it does look likely.

I wonder if that requirement could be dropped by any chance, or maybe you can think of some other workaround? Happy to run other commands if you need more information on the tablet.

**Comment by saghul** [comment]
To be honest you already used all the tricks I had in the bag, I don't know what to suggest for you to try...

**Comment by eggpi** [comment]
OK, so I ended up doing one more test: I really didn't know how to deal with the errors in debug mode, so I just copied the Play Store apk out of another (compatible) Android device and installed on the tablet with `adb install`. Everything seems works fine on the tablet now, so it does look like the main problem is just getting the app to install.

Let's just remove the autofocus requirement then? It seems like a pretty safe change to make, and we're reasonably confident it would fix the problem...
